src/main/java/com/github/onsdigital/thetrain/configuration/AppConfiguration.java

```diff
@@ -147,18 +174,63 @@ private static Path loadWebsitePathConfig() throws ConfigurationException {
     }
 
     private static int loadPublishPoolSizeConfig() throws ConfigurationException {
+        String value = System.getenv(MAX_REQUEST_SIZE_MB);
         try {
             return Integer.parseInt(System.getenv(THREAD_POOL_SIZE_ENV_KEY));
-        } catch (NumberFormatException e) {
-            throw new ConfigurationException(THREAD_POOL_SIZE_PARSE_ERR, e);
+        } catch (NumberFormatException ex) {
+            throw new ConfigurationException(formatParsingError(THREAD_POOL_SIZE_ENV_KEY, value, Integer.class), ex);
         }
     }
 
     private static int loadPortConfig() throws ConfigurationException {
+        String value = System.getenv(MAX_REQUEST_SIZE_MB);
         try {
             return Integer.parseInt(System.getenv(PORT_ENV_KEY));
         } catch (NumberFormatException e) {
-            throw new ConfigurationException(PORT_PARSE_ERR, e);
+            throw new ConfigurationException(formatParsingError(PORT_ENV_KEY, value, Integer.class), e);
+        }
+    }
+
+    private static Path createTmpFileUploadsDir() throws ConfigurationException {
+        try {
+            Path p = Files.createTempDirectory("tmp");
+            p.toFile().deleteOnExit();
+            return p;
+        } catch (Exception ex) {
+            throw new ConfigurationException("error creating tmp file uploads dir", ex);
         }
     }
-}
+
+    private static long loadMaxFileSizeConfig() throws ConfigurationException {
+        String value = System.getenv(MAX_REQUEST_SIZE_MB);
+        try {
+            return Long.parseLong(System.getenv(MAX_FILE_UPLOAD_SIZE_MB));
+        } catch (NumberFormatException ex) {
+            throw new ConfigurationException(formatParsingError(MAX_FILE_UPLOAD_SIZE_MB, value, Long.class), ex);
+        }
+    }
+
+    private static long loadMaxRequestSizeConfig() throws ConfigurationException {
+        String value = System.getenv(MAX_REQUEST_SIZE_MB);
+        try {
+            return Long.parseLong(value);
+        } catch (NumberFormatException ex) {
+            throw new ConfigurationException(formatParsingError(MAX_REQUEST_SIZE_MB, value, Long.class), ex);
+        }
+    }
+
+    private static int loadFileThresholdSizeConfig() throws ConfigurationException {
+        String value = System.getenv(MAX_REQUEST_SIZE_MB);
+        try {
+            int mb = Integer.parseInt(System.getenv(FILE_THRESHOLD_SIZE_MB));
+            return 1024 * 1024 * mb;
```

Comment:
Seems a bit inconsistent across the different config values - all of the env vars suggest a MB value, but only this function does `1024 * 1024 * mb`?

Reply:
I agree - however this is the input arg structure of the library these values are being passed into.
(long noOfBytes, long noOfBytes, int noOfMB)

Reply:
https://docs.oracle.com/javaee/7/api/javax/servlet/MultipartConfigElement.html#MultipartConfigElement-java.lang.String-long-long-int-

Reply:
https://docs.oracle.com/javaee/7/api/javax/servlet/MultipartConfigElement.html#MultipartConfigElement-java.lang.String-long-long-int-

Reply:
The thing that was throwing me was that all of the env vars have `_MB_` to suggest a MB value, but some are byte values?